DownFall/Models/Board.swift

```diff
@@ -114,55 +120,30 @@ class Board: Equatable {
         }
         
         func calculateAttacks(for entity: EntityModel, from position: TileCoord) -> [TileCoord] {
-            let attackDirections = entity.attack.directions
             let attackRange = entity.attack.range
             var affectedTiles: [TileCoord] = []
-            for direction in attackDirections {
+            
+            // TODO: Let's add a property to attacks that says if the attack goes thru targets or not
+            for attackSlope in entity.attack.attackSlope ?? [] {
                 for i in attackRange.lower...attackRange.upper {
-                    // TODO: Let's add a property to attacks that says if the attack goes thru targets or not
-                    let target = calculateTarget(in: direction, distance: i, from: position)
+                    let target = calculateTargetSlope(in: attackSlope, distance: i, from: position)
                     if isWithinBounds(target) {
                         affectedTiles.append(target)
                     }
                 }
             }
             return affectedTiles
         }
-        
-        func calculateTarget(in direction: Direction, distance i: Int, from position: TileCoord) -> TileCoord {
+
+        func calculateTargetSlope(in slopedDirection: AttackSlope, distance i: Int, from position: TileCoord) -> TileCoord {
             let (initialRow, initialCol) = position.tuple
-            let targetRow: Int
-            let targetCol: Int
-            switch direction {
-            case .north:
-                targetRow = initialRow + i
-                targetCol = initialCol
-            case .south:
-                targetRow = initialRow - i
-                targetCol = initialCol
-            case .east:
-                targetCol = initialCol + i
-                targetRow = initialRow
-            case .west:
-                targetCol = initialCol - i
-                targetRow = initialRow
-            case .northEast:
-                targetRow = initialRow + i
-                targetCol = initialCol + i
-            case .southEast:
-                targetRow = initialRow - i
-                targetCol = initialCol + i
-            case .northWest:
-                targetRow = initialRow + i
-                targetCol = initialCol - i
-            case .southWest:
-                targetRow = initialRow - i
-                targetCol = initialCol - i
-            }
             
-            return TileCoord(targetRow, targetCol)
+            // Take the initial position and calculate the target
+            // Add the slope's "up" value multiplied by the distance to the row
+            // Add the slope's "over" value multipled by the distane to the column
+            return TileCoord(initialRow + (i * slopedDirection.up), initialCol + (i * slopedDirection.over))
```

Comment:
This sucks that we copy and pasted this. Is there any way to share this logic?

Reply:
I moved some shared logic to entity.  But the rest is really logic on [[Tile]] which I didn't feel like writing an extension for